lib/puppet/provider/package/dpkg.rb

```diff
@@ -173,14 +173,10 @@ def purge
     dpkg "--purge", @resource[:name]
   end
 
-  def deprecated_hold
+  def hold
     if package_not_installed?
       self.install
     end
-    hold
-  end
-
-  def hold
```

Comment:
I'm not sure we still need to call self.install in this step, as it may run the install twice

Reply:
Yep, it seems it installs the package without doing the `self.install` here. 
I think `package_not _installed?` method can also be removed as I only found it being used here.